rules/Php83/Rector/ClassConst/AddTypeToConstRector.php

```diff
@@ -190,65 +171,20 @@ private function findValueType(Expr $expr): ?Identifier
     /**
      * @return ClassReflection[]
      */
-    private function getParents(Class_ $class): array
+    private function getParentReflections(string $className): array
     {
-        $parents = array_filter([$class->extends]);
-
-        return array_map(function (Name $name): ClassReflection {
-            if (! $name instanceof FullyQualified) {
-                throw new FullyQualifiedNameNotAutoloadedException($name);
-            }
-
-            if ($this->reflectionProvider->hasClass($name->toString())) {
-                return $this->reflectionProvider->getClass($name->toString());
-            }
-
-            throw new FullyQualifiedNameNotAutoloadedException($name);
-        }, $parents);
-    }
-
-    /**
-     * @return ClassReflection[]
-     */
-    private function getImplementations(Class_ $class): array
-    {
-        return array_map(function (Name $name): ClassReflection {
-            if (! $name instanceof FullyQualified) {
-                throw new FullyQualifiedNameNotAutoloadedException($name);
-            }
-
-            if ($this->reflectionProvider->hasClass($name->toString())) {
-                return $this->reflectionProvider->getClass($name->toString());
-            }
-
-            throw new FullyQualifiedNameNotAutoloadedException($name);
-        }, $class->implements);
-    }
-
-    /**
-     * @return ClassReflection[]
-     */
-    private function getTraits(Class_ $class): array
-    {
-        $traits = [];
-        foreach ($class->getTraitUses() as $traitUse) {
-            $traits = [...$traits, ...$traitUse->traits];
+        if (! $this->reflectionProvider->hasClass($className)) {
+            return [];
         }
 
-        return array_map(function (Name $name): ClassReflection {
-            if (! $name instanceof FullyQualified) {
-                throw new FullyQualifiedNameNotAutoloadedException($name);
-            }
-
-            if ($this->reflectionProvider->hasClass($name->toString())) {
-                return $this->reflectionProvider->getClass($name->toString());
-            }
+        $currentClassReflection = $this->reflectionProvider->getClass($className);
 
-            throw new FullyQualifiedNameNotAutoloadedException($name);
-        }, $traits);
+        return array_filter($currentClassReflection->getAncestors(), static fn (ClassReflection $classReflection): bool =>
```

Comment:
Empty parent+interface by PHPStan doesn't mean it doesn't have parent/interface, it can because they not autoloaded, it can be refactored by using existing services that verify it, but this change seems invalid